### PR TITLE
Start moving the UI/UX to LiveView

### PR DESCRIPTION
**Pull request by joshk**
The core of this PR is the transition from the older style Phoenix controllers to LiveViews for web interactions.

I've started with all Org focused pieces first. This includes the following URLs:

```
/orgs
/orgs/new
/org/:org_name
/org/:org_name/new
/org/:org_name/settings/keys
/org/:org_name/settings/keys/new
/org/:org_name/settings/users
/org/:org_name/settings/users/invite
/org/:org_name/settings/users/:user_id/edit
/org/:org_name/settings/certificates
/org/:org_name/settings/certificates/new
/org/:org_name/settings/certificates/:serial/edit
/org/:org_name/settings
```

I've also added tests for all these pages and interactions. These tests are more complete than the previous ones as forms are being filled in and submitted.

I've also included a few features along the way:
- storing who a Signing Key belongs to
- storing who invited a team member

And fixed some bugs which exist in the current UI.

### CODE REVIEW

File: lib/nerves_hub/accounts.ex [hunk 1]
```diff
@@ -224,15 +222,10 @@ defmodule NervesHub.Accounts do
...
-    product_query = from(p in Product, where: is_nil(p.deleted_at))
-
-    orgs_preload = {org_query, products: product_query}
-
     User
-    |> where([u], u.id == ^user_id)
+    |> where(id: ^user_id)
     |> Repo.exclude_deleted()
-    |> preload(orgs: ^orgs_preload)
+    |> preload(orgs: [:products])
```
Comment (lawik) [inline lib/nerves_hub/accounts.ex:228]: This seems to drop the `deleted_at` soft delete check. Is this fine? Or taken care of elsewhere?

File: lib/nerves_hub_web/channels/device_socket.ex [hunk 2]
```diff
@@ -117,16 +117,17 @@ defmodule NervesHubWeb.DeviceSocket do
...
   end
 
-  def shared_secrets_enabled?() do
-    enabled =
-      Application.get_env(:nerves_hub, __MODULE__, [])
-      |> Keyword.get(:shared_secrets, [])
-      |> Keyword.get(:enabled, false)
-
-    if enabled do
+  def shared_secrets_enabled_with_error?() do
```
Comment (lawik) [inline lib/nerves_hub_web/channels/device_socket.ex:120]: Yo dawg, we can't do `?` at the end if it doesn't strictly return a bool :)

I suggest `check_shared_secret_enabled` and return `:ok` on success.

File: lib/nerves_hub_web/components/form_helpers.ex [hunk 3]
```diff
@@ -1,5 +1,5 @@
-defmodule NervesHubWeb.OrgKeyView do
-  use NervesHubWeb, :view
+defmodule NervesHubWeb.Components.FormHelpers do
```
Comment (lawik) [inline lib/nerves_hub_web/components/form_helpers.ex:1]: This module took a big trip into a different name and role. But seems legit :)

**Review by lawik — COMMENTED** [review]
Have looked at the stuff that sticks out. Have read and skimmed a lot of it. Good job on doing all this.

File: lib/nerves_hub_web/channels/device_socket.ex (hunk 2 above)
Comment (joshk) [reply lib/nerves_hub_web/channels/device_socket.ex:120]: goooooooood point

File: lib/nerves_hub/accounts.ex (hunk 1 above)
Comment (joshk) [reply lib/nerves_hub/accounts.ex:228]: Good spotting. That should be taken case by User and Org. 

User has:
```
has_many(:orgs, through: [:org_users, :org], where: [deleted_at: nil])
```
But Org is missing a similar `where`. I'll get that fixed.

Comment (joshk) [reply lib/nerves_hub/accounts.ex:228]: oh, no, I'm wrong, I already fixed that.

File: lib/nerves_hub_web/components/form_helpers.ex (hunk 3 above)
Comment (joshk) [reply lib/nerves_hub_web/components/form_helpers.ex:1]: yeah, its kinda funny how git did this.

**Review by jjcarstens — APPROVED** [review]
Looks good enough to try. Left one comment about separating out the Elixir 1.17.1 change

File: .tool-versions
[the remark this answers is not in the document]
Comment (joshk) [reply .tool-versions:1]: done!

File: Dockerfile [hunk 4]
```diff
@@ -1,6 +1,6 @@
 ARG ELIXIR_VERSION=1.17.0
 ARG OTP_VERSION=27.0
-ARG DISTRO=noble-20240530
+ARG DISTRO=noble-20240605
```
Comment (fhunleth) [inline Dockerfile:3]: Not sure if this was intended. Elixir 1.17.1 is out now if you're going to do a Dockerfile update.

**Review by fhunleth — APPROVED** [review]

File: Dockerfile (hunk 4 above)
Comment (joshk) [reply Dockerfile:3]: https://github.com/nerves-hub/nerves_hub_web/pull/1335#pullrequestreview-2143086188

Comment (jjcarstens) [reply Dockerfile:3]: Sorry, I prob should've specified that we could probably include this docker image change wherever we're doing the Elixir 1.17 change

**Pull request merged**